docs/dag-indexer.md

```diff
@@ -0,0 +1,39 @@
+### DAG Indexer — Service Overview
+
+SubQuery-based indexer that listens to `system.remark` extrinsics and indexes DAG-PB nodes produced by `@autonomys/auto-dag-data`. Persists node metadata, links, sizes, and upload options for downstream services like File Retriever.
+
+### Flow
+
+- SubQuery runtime connects to Subspace network (`wss://rpc.mainnet.subspace.foundation/ws`).
+- For each successful `system.remark` extrinsic, decode DAG-PB, compute CID and BLAKE3, extract links and IPLD metadata.
```

Comment:
Not sure if it's worth mentioning that history seed mapping support was only in some runtimes

```suggestion
- For each successful `system.remark`, `remark_with_event`, and `history_seed` extrinsic, decode DAG-PB, compute CID and BLAKE3, extract links and IPLD metadata.
```

Reply:
This is not correct, we only listen to `system.remark`:

https://github.com/autonomys/auto-files-gateway/blob/update-docs/services/dag-indexer/project.ts#L58

Reply:
Ah, ok, so remark_with_event mappings are ignored, even if they're provided by the runtime / RPC server?

Because that's what it currently does:
https://github.com/autonomys/subspace/blob/177c4237d62efd66dc23412880ba3489468e5b04/crates/subspace-runtime/src/object_mapping.rs#L100

Previous versions of the runtime code also created mappings for history seeding (and those runtime versions are used when old blocks with history seeds are proceeded by the node):
https://github.com/autonomys/subspace/blob/d0477c588f3dd441a21407f0f493377617c9b790/crates/subspace-runtime/src/object_mapping.rs#L95

Reply:
We should index `system.remark_with_event`, let's create a ticket to address this.

Reply:
Created one at #144